### PR TITLE
fixes #0002633 - body disappear if dragged out of Part container

### PR DESCRIPTION
**Pull request by usakhelo**
Fixes the reported issue #0002633

Please see forum [discussion](http://forum.freecadweb.org/viewtopic.php?f=17&t=18975) for more information.

Thanks!
Sergo.

### CODE REVIEW

File: src/Gui/Document.cpp [hunk 1]
```diff
@@ -1476,5 +1477,22 @@ void Document::handleChildren3D(ViewProvider* viewProvider)
                 }
             }
         }
+    } else if (viewProvider && viewProvider->isDerivedFrom(ViewProviderDocumentObjectGroup::getClassTypeId())) {
```
Comment (ickby) [inline src/Gui/Document.cpp]: As we recently introduced extensions this check is not suffiecient. Everything can be a group. You need to check for ViewProvider->hasExtension(ViewProviderDocumentObjectGroup::getExtensionClassTypeId())  This returns true for all Group, GeoFeatureGroup and OriginGroup Viewproviders.

File: src/Gui/Document.cpp [hunk 2]
```diff
@@ -1476,5 +1477,22 @@ void Document::handleChildren3D(ViewProvider* viewProvider)
                 }
             }
         }
+    } else if (viewProvider && viewProvider->isDerivedFrom(ViewProviderDocumentObjectGroup::getClassTypeId())) {
+
+        ViewProviderDocumentObject* vp = static_cast<ViewProviderDocumentObject *>(viewProvider);
+        auto* ext = vp->getObject()->getExtensionByType<App::GroupExtension>();
```
Comment (ickby) [inline src/Gui/Document.cpp]: It seems you don't need the ViewProvider extension, but the DocumentObject extension. Maybe you shpould change the initial check in the else if clause fot the DocumentObject extension. If not than the check must happen here. Currently ext could be nullptr (there is a minimal chance, but coder erros could happen).

Comment (usakhelo) [inline src/Gui/Document.cpp]: I needed DocumentObject only to get children, but I've found the shorter way: ViewProvider->ClaimChildren().
Also I've added ViewProviderDocumentObjectGroup extension check inside my else if check.

**Comment by usakhelo** [comment]
I needed DocumentObject only to get object's children, but I've found the shorter way: ViewProvider->ClaimChildren().
Also I've added ViewProviderDocumentObjectGroup extension check inside my else if check.

**Comment by ickby** [comment]
The changes look good, I think it can be merged now.

**Comment by wwmayer** [comment]
Merged.

**Pull request closed without merging**